mission_control/src/mission_control.cpp

```diff
@@ -34,165 +34,100 @@
+  load_mission_sub =
+      pnh.subscribe("load_mission", 1, &MissionControlNode::loadMissionCallback, this);
+  execute_mission_sub =
+      pnh.subscribe("execute_mission", 1, &MissionControlNode::executeMissionCallback, this);
+  abort_mission_sub =
+      pnh.subscribe("abort_mission", 1, &MissionControlNode::abortMissionCallback, this);
+  query_mission_sub =
+      pnh.subscribe("query_missions", 1, &MissionControlNode::queryMissionsCallback, this);
+  remove_mission_sub =
+      pnh.subscribe("remove_missions", 1, &MissionControlNode::removeMissionsCallback, this);
 
   // Advertise all topics and services
-  pub_report_mission_load_state = node_handle.advertise<mission_control::ReportLoadMissionState>(
-      "/mngr/report_mission_load_state", 100);
-  pub_report_mission_execute_state =
-      node_handle.advertise<mission_control::ReportExecuteMissionState>(
-          "/mngr/report_mission_execute_state", 100);
-  pub_report_missions =
-      node_handle.advertise<mission_control::ReportMissions>("/mngr/report_missions", 100);
-  pub_report_heartbeat =
-      node_handle.advertise<mission_control::ReportHeartbeat>("/mngr/report_heartbeat", 100);
+  pub_report_mission_load_state =
+      pnh.advertise<mission_control::ReportLoadMissionState>("report_mission_load_state", 100);
+  pub_report_mission_execute_state = pnh.advertise<mission_control::ReportExecuteMissionState>(
+      "report_mission_execute_state", 100);
+  pub_report_missions = pnh.advertise<mission_control::ReportMissions>("report_missions", 100);
+  pub_report_heartbeat = pnh.advertise<mission_control::ReportHeartbeat>("report_heartbeat", 100);
+  pub_activate_manual_control = pnh.advertise<jaus_ros_bridge::ActivateManualControl>(
+      "/jaus_ros_bridge/activate_manual_control", 1);
 
   reportExecuteMissionStateTimer =
-      node_handle.createTimer(ros::Duration(reportExecuteMissionStateRate),
-                              &MissionControlNode::reportExecuteMissionState, this);
-  reportHeartbeatTimer = node_handle.createTimer(ros::Duration(reportHeartbeatRate),
-                                                 &MissionControlNode::reportHeartbeat, this);
-}
+      pnh.createTimer(ros::Duration(reportExecuteMissionStateRate),
+                      &MissionControlNode::reportExecuteMissionState, this);
+  reportHeartbeatTimer = pnh.createTimer(ros::Duration(reportHeartbeatRate),
+                                         &MissionControlNode::reportHeartbeat, this);
 
-MissionControlNode::~MissionControlNode()
-{
-  stop();
-
-  std::map<int, Mission*>::iterator it;
-
-  for (it = m_mission_map.begin(); it != m_mission_map.end(); it++)
-  {
-    if (it->second != NULL)
-    {
-      delete it->second;
-    }
-  }
+  executeMissionTimer = pnh.createTimer(ros::Duration(executeMissionAsynchronousRate),
+                                        &MissionControlNode::executeMissionT, this);
+  executeMissionTimer.stop();
 
-  m_mission_map.clear();
+  registerBehaviorActions();
```

Comment:
@patapufate why have a mission factory and register behavior tree node types here?